src/odemis/driver/andorshrk.py

```diff
@@ -757,6 +859,8 @@ def GetGratingInfo(self, grating):
 
         return Lines.value * 1e3, Blaze.value.decode('latin1'), Home.value, Offset.value
 
+    # don't call with reconnect, it sometimes might raise a 20201 error that's not related
+    # to a lost connection
```

Comment:
I think it should still try to reconnect, especially as it's one of the most common function. Maybe only call reconnect if failed 5 times.